File: ios/Classes/dropIn/dropInAdvancedFlow/models/ResultCode.swift
import Foundation

enum ResultCode: String, Decodable {
    case authorised = "Authorised"
    case refused = "Refused"
    case pending = "Pending"
    case cancelled = "Cancelled"
    case error = "Error"
    case received = "Received"
    case redirectShopper = "RedirectShopper"
    case identifyShopper = "IdentifyShopper"
    case challengeShopper = "ChallengeShopper"
    case presentToShopper = "PresentToShopper"
}

File: lib/src/common/model/result_code.dart
enum ResultCode {
  authenticationFinished("AuthenticationFinished"),
  authenticationNotRequired("AuthenticationNotRequired"),
  authorised("Authorised"),
  refused("Refused"),
  pending("Pending"),
  cancelled("Cancelled"),
  error("Error"),
  received("Received"),
  redirectShopper("RedirectShopper"),
  identifyShopper("IdentifyShopper"),
  challengeShopper("ChallengeShopper"),
  presentToShopper("PresentToShopper"),
  partiallyAuthorised("PartiallyAuthorised"),
  unknown("Unknown");

  final String name;

  const ResultCode(this.name);
}
